WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,182 @@
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement)
+  end,
+  SetVisible = function(self, visible)
+    if visible then
+      self:Show()
+    else
+      self:Hide()
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacement = function(self, placement)
+    local offset, offsetx, offsety = self.tick_placement, 0, 0
+    local width = self.parentTrueWidth
+    if self.tick_placement_mode == "STATIC" then
+      local percent = string.match(placement, "(%d+)%%")
+      if percent then
+        offset = (tonumber(percent) / 100) * width
+      else
+        local pixels = width / (self.parent.state.duration or self.parent.state.total or 1)
+        offset = tonumber(placement) * pixels
+      end
+    end
+
+    if self.parentInverse ~= ((self.parentOrientation == "HORIZONTAL_INVERSE") or (self.parentOrientation == "VERTICAL")) then
+      offset = -offset
+    end
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      offsety = offset
+    else
+      offsetx = offset
+    end
+    local side = self.parentInverse and auraBarTrueLeftInverse or auraBarTrueLeft
+
+    self:ClearAllPoints()
+    self:SetPoint("CENTER", self.parent.bar, side[self.parentOrientation], offsetx, offsety)
+  end,
+  SetTickWidth = function(self, width)
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      self:SetHeight(width)
+    else
+      self:SetWidth(width)
+    end
+  end,
+  SetTickHeight = function(self, height)
+    if self.automatic_height then height = self.parentTrueHeight end
+    if (self.parentOrientation == "VERTICAL") or (self.parentOrientation == "VERTICAL_INVERSE") then
+      self:SetWidth(height)
+    else
+      self:SetHeight(height)
+    end
+  end,
+}
+
+local function modify(parent, region, parentData, data, first)
+
+  region:SetParent(parent)
+  region.parentOrientation = parentData.orientation
+  region.parentInverse = parentData.inverse
+  if (region.parentOrientation == "VERTICAL") or (region.parentOrientation == "VERTICAL_INVERSE") then
+    region.parentTrueHeight, region.parentTrueWidth = parent.bar:GetRealSize()
```

Comment:
This is not good enough. The aurabar can resize itself via conditions/animations.

You should get the size in SetTickReplacement, and in modify, do:
region:SetAllPoints(parent)
region:SetScript("OnSizeChanged", function(self) self:SetTickPlacement(...) end

Also to your great annoyance, AuraBars can change their orientation via SetOrientation.
You need to introduce a new "event" that is Notified in SetOrientation and subscribed too.